src/input.rs

```diff
@@ -269,44 +271,41 @@ impl<'a, A: ActionContext + 'a> Processor<'a, A> {
         self.ctx.mouse_mut().x = x;
         self.ctx.mouse_mut().y = y;
 
-        let size_info = self.ctx.size_info();
-        if let Some(point) = size_info.pixels_to_coords(x as usize, y as usize) {
+        if let Some((point, cell_side)) = self.get_mouse_pos() {
             let prev_line = mem::replace(&mut self.ctx.mouse_mut().line, point.line);
             let prev_col = mem::replace(&mut self.ctx.mouse_mut().column, point.col);
 
-            let cell_x = (x as usize - size_info.padding_x as usize) % size_info.cell_width as usize;
-            let half_cell_width = (size_info.cell_width / 2.0) as usize;
-
-            let cell_side = if cell_x > half_cell_width {
-                Side::Right
-            } else {
-                Side::Left
-            };
             self.ctx.mouse_mut().cell_side = cell_side;
 
             let motion_mode = TermMode::MOUSE_MOTION | TermMode::MOUSE_DRAG;
-            if self.ctx.mouse_mut().left_button_state == ElementState::Pressed
+            if self.ctx.mouse().left_button_state == ElementState::Pressed
                 && (
                     modifiers.shift
                     || !self.ctx.terminal_mode().intersects(TermMode::MOUSE_REPORT_CLICK | motion_mode)
                 )
             {
-                self.ctx.update_selection(Point {
-                    line: point.line,
-                    col: point.col
-                }, cell_side);
+                if self.ctx.selection_started() {
+                    self.ctx.update_selection(Point {
+                        line: point.line,
+                        col: point.col
+                    }, cell_side);
+                } else {
+                    let pos = self.ctx.mouse().selection_start_pos;
+                    let side = self.ctx.mouse().selection_start_side;
+                    self.ctx.update_selection(pos, side);
```

Comment:
Really small nitpick, but we're using the words `point` and `pos` interchangeably. Might be nice to clean that up. That's a larger refactor, I understand if you don't want to make it.

Reply:
Yeah while writing this code I wasn't entirely sure if I should just rename `selection_start_pos` to `selection_start_point`. In theory it's both the same thing because a `Point` is just a position somewhere in the grid.

I think I just went for `pos` because it's shorter tbh, if you already got confused by this it might make sense to change it just for consistency.